### PR TITLE
ts-738 remove redundant null checks

### PR DESCRIPTION
**Pull request by tomskikh**

### CODE REVIEW

File: t-streams/src/main/scala/com/bwsw/tstreams/agents/consumer/ConsumerTransaction.scala [hunk 1]
```diff
@@ -118,8 +115,8 @@ class ConsumerTransaction(partition: Int,
   /**
     * @return All consumed transaction
     */
-  def getAll = this.synchronized {
-    if (consumer == null)
```
Comment (bwsw) [inline t-streams/src/main/scala/com/bwsw/tstreams/agents/consumer/ConsumerTransaction.scala]: Why do you think it's not required?

File: t-streams/src/main/scala/com/bwsw/tstreams/agents/consumer/ConsumerTransaction.scala [hunk 2]
```diff
@@ -80,9 +77,9 @@ class ConsumerTransaction(partition: Int,
   /**
     * @return Next piece of data from current transaction
     */
-  def next() = this.synchronized {
+  def next(): Array[Byte] = this.synchronized {
 
-    if (consumer == null)
```
Comment (bwsw) [inline t-streams/src/main/scala/com/bwsw/tstreams/agents/consumer/ConsumerTransaction.scala]: Why do you think it's not required?

File: tstreams-transaction-server/src/main/scala/com/bwsw/tstreamstransactionserver/netty/server/multiNode/TestCommonCheckpointGroupServer.scala [hunk 3]
```diff
@@ -81,11 +81,7 @@ class TestCommonCheckpointGroupServer(authenticationOpts: AuthenticationOptions,
```
Comment (bwsw) [inline tstreams-transaction-server/src/main/scala/com/bwsw/tstreamstransactionserver/netty/server/multiNode/TestCommonCheckpointGroupServer.scala]: rename to CommonCheckpointGroupTestServer please.

File: tstreams-transaction-server/src/main/scala/com/bwsw/tstreamstransactionserver/netty/server/singleNode/TestSingleNodeServer.scala [hunk 4]
```diff
@@ -79,11 +79,7 @@ class TestSingleNodeServer(authenticationOpts: AuthenticationOptions,
```
Comment (bwsw) [inline tstreams-transaction-server/src/main/scala/com/bwsw/tstreamstransactionserver/netty/server/singleNode/TestSingleNodeServer.scala]: rename to SingleNodeTestServer please.

Comment (tomskikh) [reply tstreams-transaction-server/src/main/scala/com/bwsw/tstreamstransactionserver/netty/server/singleNode/TestSingleNodeServer.scala]: Renamed to SingleNodeTestingServer according to https://github.com/bwsw/t-streams/pull/747#discussion_r164063063.

File: t-streams/src/main/scala/com/bwsw/tstreams/agents/consumer/ConsumerTransaction.scala [hunk 5]
```diff
@@ -47,11 +47,8 @@ class ConsumerTransaction(partition: Int,
   def consumer: Consumer =
     _consumer.getOrElse(throw new IllegalStateException("The transaction is not yet attached to consumer"))
 
-  def attach(c: Consumer) = {
-    if (c == null)
```
Comment (placccebo) [inline t-streams/src/main/scala/com/bwsw/tstreams/agents/consumer/ConsumerTransaction.scala]: I suppose, that it's required (with reference to bwsw review) but if we think then we'll realize that there are a lot of methods which parameters could be null

File: tstreams-transaction-server/src/main/scala/com/bwsw/tstreamstransactionserver/netty/server/multiNode/TestCommonCheckpointGroupServer.scala (hunk 3 above)
Comment (tomskikh) [reply tstreams-transaction-server/src/main/scala/com/bwsw/tstreamstransactionserver/netty/server/multiNode/TestCommonCheckpointGroupServer.scala]: Renamed to CommonCheckpointGroupTestingServer.

File: t-streams/src/main/scala/com/bwsw/tstreams/agents/consumer/ConsumerTransaction.scala (hunk 2 above)
Comment (tomskikh) [reply t-streams/src/main/scala/com/bwsw/tstreams/agents/consumer/ConsumerTransaction.scala]: I replaced it by isAttached. It is the same.

File: t-streams/src/main/scala/com/bwsw/tstreams/agents/consumer/ConsumerTransaction.scala (hunk 1 above)
Comment (tomskikh) [reply t-streams/src/main/scala/com/bwsw/tstreams/agents/consumer/ConsumerTransaction.scala]: I replaced it by isAttached. It is the same.

File: t-streams/src/main/scala/com/bwsw/tstreams/agents/consumer/ConsumerTransaction.scala (hunk 5 above)
Comment (tomskikh) [reply t-streams/src/main/scala/com/bwsw/tstreams/agents/consumer/ConsumerTransaction.scala]: That method is used only by Consumer, and Consumer attachs itself to transaction (```transaction.attach(this)```). So, method's argument cannot be null.

File: t-streams/src/main/scala/com/bwsw/tstreams/common/SpareServerSocketLookupUtility.scala [hunk 6]
```diff
@@ -19,30 +19,30 @@
...
+    val result = Try {
+      serverSocket = Some(new ServerSocket(port, 1, InetAddress.getByName(hostOrIp)))
+      serverSocket.foreach(_.setReuseAddress(true))
+      serverSocket.foreach(_.close())
+    }
+
+    result match {
+      case Success(_) => true
+      case Failure(_) =>
+        serverSocket.foreach(_.close())
```
[the remark this answers is not in the document]
Comment (tomskikh) [reply t-streams/src/main/scala/com/bwsw/tstreams/common/SpareServerSocketLookupUtility.scala]: Fixed

File: t-streams/src/main/scala/com/bwsw/tstreams/agents/consumer/ConsumerTransaction.scala (hunk 5 above)
Comment (placccebo) [reply t-streams/src/main/scala/com/bwsw/tstreams/agents/consumer/ConsumerTransaction.scala]: ok))

**Pull request merged**